### PR TITLE
Include TVstock variant

### LINKED ISSUE #47: leanback support
Can you pleas add support to build for leanback
' ' '
ric@ric-ubuntu:~/androidx86tv$ lunch android_x86_64-userdebug
vendor/google/build/config.mk:34: **\* GAPPS_VARIANT leanback was not found. Use of one of pico,nano,micro,mini,full,stock,super.  Stop.
' ' '

### PR DESCRIPTION
**Pull request by kyvaith**
Resolves  #47 and #118.

### CODE REVIEW

**Comment by Blystad** [comment]
Hey,

Thanks for adding support for TV. One suggestion though, what about
defaulting to PRESIGNED as before, but allow that to be overwritten. I
think this might be doable by changing the LOCAL_CERTIFICATE in the
prebuilt_apk.mk to ?= Instead of :=

Alternatively, just do a check if local cert is empty, and if it is then
overwrite it with the PRESIGNED default.


On 3 Nov 2017 22:13, "Tomasz Witke" <notifications@github.com> wrote:

Resolves #47 <https://github.com/opengapps/aosp_build/issues/47> and #118
<https://github.com/opengapps/aosp_build/issues/118>.
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/opengapps/aosp_build/pull/158
Commit Summary

   - Added new TVstock variant
   - Change APK's signing rules to grant system apps 'dangerous' permissions
   - Delete global signing rule
   - Maybe we can't?
   - Update
   - Add missing signing rules
   - Correct signing rules accorting to https://github.com/opengapps/
   aosp_build/issues/145#issuecomment-322046367
   <https://github.com/opengapps/aosp_build/issues/145#issuecomment-322046367>
   - Merge https://github.com/opengapps/opengapps/pull/543 and correct
   https://github.com/opengapps/opengapps/pull/508
   <https://github.com/opengapps/opengapps/pull/508>
   - No need to include old BugReportSender

File Changes

   - *M* config.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-0> (2)
   - *M* core/definitions.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-1> (1)
   - *M* core/prebuilt_apk.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-2> (2)
   - *A* modules/AndroidMediaShell/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-3> (9)
   - *M* modules/AndroidPlatformServices/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-4> (1)
   - *A* modules/AtvRemoteService/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-5> (30)
   - *A* modules/AtvWidget/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-6> (8)
   - *A* modules/Backdrop/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-7> (8)
   - *M* modules/Books/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-8> (1)
   - *A* modules/BugReportSender/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-9> (8)
   - *M* modules/CalculatorGoogle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-10> (1)
   - *M* modules/CalendarGooglePrebuilt/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-11> (1)
   - *A* modules/CanvasPackageInstaller/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-12> (9)
   - *M* modules/Chrome/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-13> (1)
   - *M* modules/CloudPrint2/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-14> (1)
   - *A* modules/ConfigUpdater/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-15> (9)
   - *M* modules/DMAgent/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-16> (1)
   - *M* modules/Drive/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-17> (1)
   - *M* modules/EditorsDocs/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-18> (1)
   - *M* modules/EditorsSheets/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-19> (1)
   - *M* modules/EditorsSlides/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-20> (1)
   - *M* modules/FaceLock/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-21> (1)
   - *M* modules/Feedback/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-22> (1)
   - *M* modules/FitnessPrebuilt/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-23> (1)
   - *M* modules/GCS/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-24> (1)
   - *A* modules/GamepadPairingService/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-25> (9)
   - *A* modules/GlobalKeyInterceptor/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-26> (9)
   - *M* modules/GoogleBackupTransport/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-27> (1)
   - *M* modules/GoogleCalendarSyncAdapter/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-28> (1)
   - *M* modules/GoogleCamera/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-29> (1)
   - *M* modules/GoogleContacts/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-30> (1)
   - *M* modules/GoogleContactsSyncAdapter/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-31> (1)
   - *M* modules/GoogleDialer/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-32> (1)
   - *M* modules/GoogleEarth/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-33> (1)
   - *M* modules/GoogleExtServices/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-34> (1)
   - *M* modules/GoogleExtShared/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-35> (1)
   - *M* modules/GoogleFeedback/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-36> (1)
   - *M* modules/GoogleHindiIME/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-37> (1)
   - *M* modules/GoogleHome/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-38> (1)
   - *M* modules/GoogleJapaneseInput/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-39> (1)
   - *M* modules/GoogleLoginService/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-40> (1)
   - *M* modules/GoogleOneTimeInitializer/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-41> (1)
   - *M* modules/GooglePackageInstaller/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-42> (1)
   - *M* modules/GooglePartnerSetup/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-43> (1)
   - *M* modules/GooglePinyinIME/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-44> (1)
   - *M* modules/GooglePrintRecommendationService/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-45> (1)
   - *M* modules/GoogleServicesFramework/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-46> (1)
   - *M* modules/GoogleTTS/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-47> (1)
   - *M* modules/GoogleVrCore/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-48> (1)
   - *M* modules/GoogleZhuyinIME/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-49> (1)
   - *M* modules/Hangouts/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-50> (1)
   - *A* modules/Katniss/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-51> (9)
   - *M* modules/KoreanIME/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-52> (1)
   - *A* modules/LandscapeWallpaper/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-53> (8)
   - *M* modules/LatinImeGoogle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-54> (1)
   - *A* modules/LeanbackIme/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-55> (8)
   - *A* modules/LeanbackLauncher/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-56> (9)
   - *M* modules/Maps/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-57> (1)
   - *M* modules/Music2/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-58> (1)
   - *A* modules/Music2Pano/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-59> (8)
   - *M* modules/Newsstand/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-60> (1)
   - *A* modules/NoTouchAuthDelegate/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-61> (8)
   - *A* modules/Overscan/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-62> (9)
   - *M* modules/Phonesky/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-63> (1)
   - *M* modules/Photos/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-64> (1)
   - *M* modules/PixelLauncher/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-65> (1)
   - *M* modules/PixelLauncherIcons/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-66> (1)
   - *M* modules/PlayGames/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-67> (1)
   - *M* modules/PlusOne/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-68> (1)
   - *M* modules/PrebuiltBugle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-69> (1)
   - *M* modules/PrebuiltDeskClockGoogle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-70> (1)
   - *M* modules/PrebuiltExchange3Google/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-71> (1)
   - *M* modules/PrebuiltGmail/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-72> (1)
   - *M* modules/PrebuiltGmsCore/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-73> (1)
   - *M* modules/PrebuiltGmsCoreInstantApps/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-74> (1)
   - *A* modules/PrebuiltGmsCorePano/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-75> (9)
   - *M* modules/PrebuiltKeep/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-76> (1)
   - *M* modules/PrebuiltNewsWeather/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-77> (1)
   - *A* modules/RemoteControlService/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-78> (9)
   - *A* modules/SecondScreenSetup/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-79> (8)
   - *A* modules/SecondScreenSetupAuthBridge/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-80> (8)
   - *M* modules/SetupWizard/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-81> (1)
   - *A* modules/SetupWraith/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-82> (9)
   - *M* modules/Street/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-83> (1)
   - *A* modules/TV/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-84> (9)
   - *M* modules/TagGoogle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-85> (1)
   - *M* modules/TranslatePrebuilt/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-86> (1)
   - *A* modules/Tubesky/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-87> (9)
   - *M* modules/Turbo/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-88> (1)
   - *M* modules/Tycho/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-89> (1)
   - *M* modules/Velvet/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-90> (1)
   - *M* modules/Videos/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-91> (1)
   - *A* modules/VideosPano/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-92> (8)
   - *M* modules/Wallet/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-93> (1)
   - *M* modules/Wallpapers/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-94> (1)
   - *M* modules/WebViewGoogle/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-95> (1)
   - *M* modules/YouTube/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-96> (1)
   - *A* modules/YouTubeLeanback/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-97> (8)
   - *M* modules/talkback/Android.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-98> (1)
   - *M* opengapps-packages.mk
   <https://github.com/opengapps/aosp_build/pull/158/files#diff-99> (59)

Patch Links:

   - https://github.com/opengapps/aosp_build/pull/158.patch
   - https://github.com/opengapps/aosp_build/pull/158.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/opengapps/aosp_build/pull/158>, or mute the thread

File: modules/AtvRemoteService/Android.mk [hunk 1]
```diff
@@ -0,0 +1,30 @@
+LOCAL_PATH := .
+include $(CLEAR_VARS)
+include $(GAPPS_CLEAR_VARS)
+LOCAL_CERTIFICATE := platform
+LOCAL_MODULE := AtvRemoteService
+ifneq ($(filter $(call get-allowed-api-levels),24),)
```
Comment (jamuir) [inline modules/AtvRemoteService/Android.mk]: filter parameters should be swapped.

File: opengapps-packages.mk [hunk 2]
```diff
@@ -130,6 +130,45 @@ endif # end mini
 endif # end micro
 endif # end nano
 
+ifneq ($(filter $(TARGET_GAPPS_VARIANT),tvstock),)
```
Comment (jamuir) [inline opengapps-packages.mk]: same here; filter parameters should be swapped.

File: opengapps-packages.mk [hunk 3]
```diff
@@ -130,6 +130,45 @@ endif # end mini
...
+    Music2Pano \
+    CanvasPackageInstaller \
+    PlayGames \
+    Katniss \
+    AtvWidget \
+    YouTubeLeanback \
+    SetupWraith \
+    AtvRemoteService
+
+ifneq ($(filter $(call get-allowed-api-levels),24),)
```
Comment (jamuir) [inline opengapps-packages.mk]: here, too.

File: opengapps-packages.mk [hunk 4]
```diff
@@ -149,6 +188,8 @@ PRODUCT_PACKAGES += \
     WebViewGoogle
 endif
 
+ifneq ($(filter $(TARGET_GAPPS_VARIANT),pico),) # Not on tvstock
```
Comment (jamuir) [inline opengapps-packages.mk]: here, too.

also, is the comment accurate?  how is "pico" related to "tvstock"?

Comment (kyvaith) [reply opengapps-packages.mk]: tv stock does not include pico variant. Everything else include pico.

**Comment by kyvaith** [comment]
@Blystad -  changing the LOCAL_CERTIFICATE in the prebuilt_apk.mk to ?= Instead of := is a great idea, but would it work? I'm also uncomfortable with so many files changed by this pull request.

**Comment by Blystad** [comment]
I'm actually unsure if ?= Would work, but checking if LOCAL_CERTIFICATE is
empty, and if it is then setting it to PRESIGNED would for sure work (using
ifeq)

**Comment by mukgu** [comment]
hello kyvaith,
i play youtube failed Loading not Show video and sound....how to fix.thank man
I'm download file iso from here https://mega.nz/#!xgcFkTBR!hxEBZDBbuW5Lx6-YxVInK2gbi7zH1es83FtgrX-Zru8

**Comment by jamuir** [comment]
@kyvaith :  it would be great to add tvstock support.

Can you give an update on the status of your PR?

**Comment by npjohnson** [comment]
Do not resigned the apps platform.

This ruins the ability to update and causes APK signature mismatches.

GApps should contain no platform apps, those should be per-device vendor side.

**Comment by supergps1** [comment]
Sorry to intrude, not sure where else to ask this. Please point me to a correct forum for this question.

This PR doesn't work well with current repo as of 20May2019.

I am trying to build including tvstock for Hikey960, with aosp-master

I manually downloaded aosp_build, arm, arm64 and placed them in vendor/opengapps/build and vendor/opengapps/sources, and modified files under build/ manually to include all changes specified in this PR.

I receive the error "Unable to find an architecture compatible APK for package..."  for modules. if i remove a module from modules/, I get this error for next module and so on. Admittedly, I didn't check for all modules, only six modules, and then I gave up.

Any help and guidance is appreciated.

**Pull request closed without merging**